kubernetes.go

```diff
@@ -91,12 +94,22 @@ type KubernetesNodePoolUpdateRequest struct {
 	Tags  []string `json:"tags,omitempty"`
 }
 
-// KubernetesNodePoolRecycleNodesRequest represents a request to recycle a set of
-// nodes in a node pool. This will recycle the nodes by ID.
+// KubernetesNodePoolRecycleNodesRequest is DEPRECATED please use DeleteNode
 type KubernetesNodePoolRecycleNodesRequest struct {
 	Nodes []string `json:"nodes,omitempty"`
 }
 
+// KubernetesNodeDeleteRequest represents a request to delete a specific node in a node pool.
+type KubernetesNodeDeleteRequest struct {
+	// DecrementNodePoolSize whether to decrease the node pool size at the same time as deleting the node.
+	// If nil, it will use the default behavior
+	DecrementNodePoolSize *bool `json:"decrement_node_pool_size,omitempty"`
+
+	// Drain specifies whether to drain the node before deleting it.
+	// If nil, it will use the default behavior. You can't set DecrementNodePoolSize and Drain to true at the same time.
```

Comment:
I realize this behavior isn't dictated by godo, but is there a reason we're not draining the nodes before deleting them?  It still seems reasonable to reschedule their jobs to the rest of the cluster.